### PR TITLE
update the dependencies for Java Feign API clients to the latest vers…

### PR DESCRIPTION
**Pull request by zhenjun115**
#3051

### CODE REVIEW

File: modules/swagger-codegen/src/main/resources/Java/libraries/feign/build.gradle.mustache [hunk 1]
```diff
@@ -96,7 +96,7 @@ if(hasProperty('target') && target == 'android') {
 ext {
     swagger_annotations_version = "1.5.8"
     jackson_version = "2.7.5"
-    feign_version = "8.16.0"
+    feign_version = "8.17.0"
     junit_version = "4.12"
     oltu_version = "1.0.1"
```
Comment (wing328) [inline modules/swagger-codegen/src/main/resources/Java/libraries/feign/build.gradle.mustache]: For `oltu version`, can we upgrade to 1.0.2?

https://mvnrepository.com/artifact/org.apache.oltu.oauth2/org.apache.oltu.oauth2.client

File: modules/swagger-codegen/src/main/resources/Java/libraries/feign/pom.mustache [hunk 2]
```diff
@@ -161,7 +161,7 @@
     <maven.compiler.source>${java.version}</maven.compiler.source>
     <maven.compiler.target>${java.version}</maven.compiler.target>
     <swagger-core-version>1.5.8</swagger-core-version>
```
Comment (wing328) [inline modules/swagger-codegen/src/main/resources/Java/libraries/feign/pom.mustache]: For swagger core version, I believe the latest stable version is 1.5.9.

**Comment by wing328** [comment]
@zhenjun115 overall it looks good.

I left you some comments about `oltu` version. Please take a look.

**Comment by zhenjun115** [comment]
ok, i will try to update it later.

File: modules/swagger-codegen/src/main/resources/Java/libraries/feign/pom.mustache (hunk 2 above)
Comment (zhenjun115) [inline modules/swagger-codegen/src/main/resources/Java/libraries/feign/pom.mustache]: will check it later also

Comment (zhenjun115) [inline modules/swagger-codegen/src/main/resources/Java/libraries/feign/pom.mustache]: @wing328 In your file, feign client still use 1.5.8. please have a check, https://github.com/swagger-api/swagger-codegen/blob/master/samples/client/petstore/java/feign/pom.xml#L163

Comment (zhenjun115) [inline modules/swagger-codegen/src/main/resources/Java/libraries/feign/pom.mustache]: @wing328 In your file, feign client still use 1.5.8. please have a check, https://github.com/swagger-api/swagger-codegen/blob/master/samples/client/petstore/java/feign/pom.xml#L163

Comment (wing328) [inline modules/swagger-codegen/src/main/resources/Java/libraries/feign/pom.mustache]: @zhenjun115  Yes, it's. That's why I want you to help upgrade it to the latest version.

In other words, I want you to upgrade all dependencies to the latest stable version.

File: modules/swagger-codegen/src/main/resources/Java/libraries/feign/build.gradle.mustache (hunk 1 above)
Comment (zhenjun115) [inline modules/swagger-codegen/src/main/resources/Java/libraries/feign/build.gradle.mustache]: @wing328 it won't work with oltu 1.0.2. I will check why it won't work later.

File: modules/swagger-codegen/src/main/resources/Java/libraries/feign/pom.mustache (hunk 2 above)
Comment (zhenjun115) [inline modules/swagger-codegen/src/main/resources/Java/libraries/feign/pom.mustache]: @wing328 ok

**Comment by wing328** [comment]
I've updated the swagger core version so no worry.

Thanks again for the contribution.

**Pull request merged**